torch/lib/c10d/sequence_num.cpp

```diff
@@ -0,0 +1,65 @@
+#include <ATen/ThreadLocalState.h>
+#include <c10d/sequence_num.hpp>
+
+#include <c10/util/Logging.h>
+
+namespace c10d {
+SequenceNum::SequenceNum() : num_(kUnsetSeqNum), set_(false) {}
+
+SequenceNum::SequenceNum(const uint64_t num) : num_(num), set_(true) {}
```

Comment:
You can just use `std::optional<int>` to represent `num_` and no need for the field `set_`.